dbms/src/Functions/tests/gtest_substring.cpp

```diff
@@ -27,9 +27,160 @@ class SubString : public DB::tests::FunctionTest
+            sub_string.executeFunction(
+                "substringUTF8",
+                createColumn<Nullable<String>>(
+                    {"www.pingcap.com",
+                     "ww.pingcap.com",
+                     "w.pingcap.com",
+                     ".pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     {}}),
+                createColumn<T1>({-5, 1, 3, -3, 8, 2, -100, 0, 2}),
+                createColumn<T2>({4, 4, 7, 4, 5, -5, 2, 3, 6})));
+    }
+};
+
+template <typename T1, typename T2>
+class TestNullableUnsigned
+{
+public:
+    static void operator()(SubString & sub_string)
+    {
+        ASSERT_COLUMN_EQ(
+            createColumn<Nullable<String>>({"p.co", "ww.p", "pingcap", "com", ".com", "", "", {}, {}, {}}),
+            sub_string.executeFunction(
+                "substringUTF8",
+                createColumn<Nullable<String>>(
+                    {"www.pingcap.com",
+                     "ww.pingcap.com",
+                     "w.pingcap.com",
+                     ".pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     {},
+                     "pingcap",
+                     "pingcap"}),
+                createColumn<T1>({11, 1, 3, 10, 8, 2, 0, 9, {}, 7}),
+                createColumn<T2>({4, 4, 7, 4, 5, 0, 3, 6, 1, {}})));
+    }
+};
+
+template <typename T1, typename T2>
+class TestUnsigned
+{
+public:
+    static void operator()(SubString & sub_string)
+    {
+        ASSERT_COLUMN_EQ(
+            createColumn<Nullable<String>>({"p.co", "ww.p", "pingcap", "com", ".com", "", "", {}}),
+            sub_string.executeFunction(
+                "substringUTF8",
+                createColumn<Nullable<String>>(
+                    {"www.pingcap.com",
+                     "ww.pingcap.com",
+                     "w.pingcap.com",
+                     ".pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     "pingcap.com",
+                     {}}),
+                createColumn<T1>({11, 1, 3, 10, 8, 2, 0, 2}),
+                createColumn<T2>({4, 4, 7, 4, 5, 0, 3, 1})));
+    }
+};
+
+template <typename T1, typename T2>
+class TestConstPos
+{
+public:
+    static void operator()(SubString & sub_string)
+    {
+        ASSERT_COLUMN_EQ(
+            createColumn<Nullable<String>>({"w", "ww", "w.p", ".pin"}),
+            sub_string.executeFunction(
+                "substringUTF8",
+                createColumn<Nullable<String>>({"www.pingcap.com", "ww.pingcap.com", "w.pingcap.com", ".pingcap.com"}),
+                createConstColumn<T1>(4, 1),
+                createColumn<T2>({1, 2, 3, 4})));
+    }
+};
+
+template <typename T1, typename T2>
+class TestConstLength
```

Comment:
Please make sure we cover const nullable cases.

Reply:
`TestAllIntTypes` and `TestAllUIntTypes` consist of nullable types.